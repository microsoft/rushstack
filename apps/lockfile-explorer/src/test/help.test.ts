// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

import { execSync } from 'child_process';

describe('CLI Tool Tests', () => {
  it('should display help for "lockfile-explorer --help"', () => {
    const startOutput = execSync('node lib/start-explorer.js --help').toString();
    expect(startOutput).toMatchSnapshot();
  });

  it('should display help for "lockfile-lint --help"', () => {
    const lintOutput = execSync('node lib/start-lint.js --help').toString();
    expect(lintOutput).toMatchSnapshot();
  });

  it('should display help for "lockfile-lint init --help"', () => {
    const lintOutput = execSync('node lib/start-lint.js init --help').toString();
    expect(lintOutput).toMatchSnapshot();
  });

  it('should display help for "lockfile-lint check --help"', () => {
    const lintOutput = execSync('node lib/start-lint.js check --help').toString();
    expect(lintOutput).toMatchSnapshot();
  });
});
