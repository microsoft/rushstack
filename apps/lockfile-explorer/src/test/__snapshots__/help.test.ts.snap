// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI Tool Tests should display help for "lockfile-explorer --help" 1`] = `
"usage: lockfile-explorer [-h] [-d] [--subspace SUBSPACE_NAME]

Lockfile Explorer is a desktop app for investigating and solving version 
conflicts in a PNPM workspace.

Optional arguments:
  -h, --help            Show this help message and exit.
  -d, --debug           Show the full call stack if an error occurs while 
                        executing the tool
  --subspace SUBSPACE_NAME
                        Specifies an individual Rush subspace to check. The 
                        default value is \\"default\\".

[1mFor detailed help about a specific command, use: lockfile-explorer 
<command> -h[22m
"
`;

exports[`CLI Tool Tests should display help for "lockfile-lint --help" 1`] = `
"usage: lockfile-lint [-h] <command> ...

Lockfile Lint applies configured policies to find and report dependency 
issues in your PNPM workspace.

Positional arguments:
  <command>
    init      Create a new lockfile-lint.json config file
    check     Check and report dependency issues in your workspace

Optional arguments:
  -h, --help  Show this help message and exit.

[1mFor detailed help about a specific command, use: lockfile-lint <command> 
-h[22m
"
`;

exports[`CLI Tool Tests should display help for "lockfile-lint check --help" 1`] = `
"usage: lockfile-lint check [-h]

This command applies the policies that are configured in lockfile-lint.json, 
reporting any problems found in your PNPM workspace.

Optional arguments:
  -h, --help  Show this help message and exit.
"
`;

exports[`CLI Tool Tests should display help for "lockfile-lint init --help" 1`] = `
"usage: lockfile-lint init [-h]

This command initializes a new lockfile-lint.json config file. The created 
template file includes source code comments that document the settings.

Optional arguments:
  -h, --help  Show this help message and exit.
"
`;
